messagevalidator.js

```diff
@@ -0,0 +1,60 @@
+// Sets formatting requirements for json messages from clients.
+// Import and call validate_message to check whether a message
+// meets the requirements.
+
+// Items in the json are always strings and may need to be coerced
+// to other types to check validity. To check if an item is a string,
+// supply "string" as a value. If the item ought to be anything else,
+// supply a function which checks whether the appropriate coercion
+// is valid.
+
+message_formats = {
+    "set_name" : {
+            "player_name": "string",
+            "version" : "string"},
+        "join_room" : {
```

Comment:
the spacing looks different for set_name from the rest. tabs/spaces and github display?